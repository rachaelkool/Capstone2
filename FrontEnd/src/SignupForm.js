import React, { useState } from "react";
import { useHistory } from "react-router-dom";


function SignupForm({ signup }) {
    const history = useHistory();

    const [formData, setFormData] = useState({
        empId: "",
        password: "",
        firstName: "",
        lastName: ""
    });

    const [formErrors, setFormErrors] = useState([]);

    async function handleSubmit(e) {
        e.preventDefault();
        let result = await signup(formData);
        if (result.success) {
        history.push("/");
        } else {
        setFormErrors(result.errors);
        }
    }

    function handleChange(e) {
        const { name, value } = e.target;
        setFormData(data => ({ ...data, [name]: value }));
    }

    function handleChangeEmpId(e) {
        const { name, value } = e.target;
        if (isNaN(value)) {
            alert('Employee ID cannot contain letters.');
            return;
        }
        setFormData(data => ({ ...data, [name]: +value }));
    }

    return (
        <div>
            <h4>Sign Up</h4>
            <form onSubmit={handleSubmit}>
                <div>
                <label htmlFor="empId">Employee ID:</label>
                <input
                    name="empId"
                    value={formData.empId}
                    onChange={handleChangeEmpId}
                    autoComplete="empId"
                    required
                />
                </div>
                <div>
                <label htmlFor="password">Password:</label>
                <input
                    type="password"
                    name="password"
                    value={formData.password}
                    onChange={handleChange}
                    autoComplete="password"
                    required
                />
                </div>
                <div>
                <label htmlFor="firstName">First Name:</label>
                <input
                    name="firstName"
                    value={formData.firstName}
                    onChange={handleChange}
                />
                </div>
                <div>
                <label htmlFor="lastName">Last Name:</label>
                <input
                    name="lastName"
                    value={formData.lastName}
                    onChange={handleChange}
                />
                </div>
                
                <button>Submit</button>
            </form>
            <p>{formErrors}</p>
        </div>
    );
}


export default SignupForm;