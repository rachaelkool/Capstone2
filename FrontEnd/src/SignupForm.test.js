import React from "react";
import { render } from "@testing-library/react";
import SignupForm from "./SignupForm";


it("renders without crashing", function() {
    render(<SignupForm />);
});
  
it("matches snapshot", function() {
    const { asFragment } = render(<SignupForm />);
    expect(asFragment()).toMatchSnapshot();
});