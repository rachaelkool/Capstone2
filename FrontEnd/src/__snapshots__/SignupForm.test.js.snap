// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="logged-out-form"
  >
    <div
      class="form"
    >
      <h4>
        Sign Up
      </h4>
      <form
        class="ui form"
      >
        <div
          class="field"
        >
          <label
            for="empId"
          >
            Employee ID:
          </label>
          <input
            autocomplete="empId"
            name="empId"
            required=""
            value=""
          />
        </div>
        <div
          class="field"
        >
          <label
            for="password"
          >
            Password:
          </label>
          <input
            autocomplete="password"
            name="password"
            required=""
            type="password"
            value=""
          />
        </div>
        <div
          class="field"
        >
          <label
            for="firstName"
          >
            First Name:
          </label>
          <input
            name="firstName"
            value=""
          />
        </div>
        <div
          class="field"
        >
          <label
            for="lastName"
          >
            Last Name:
          </label>
          <input
            name="lastName"
            value=""
          />
        </div>
        <br />
        <button>
          Submit
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
